src/internal/collection/README.md

```diff
@@ -0,0 +1,103 @@
+# Collections
+
+Collections are a library for storing protobufs in a persistent database with a simple interface for reading, writing, and watching for changes.  There are two existing implementations, one for etcd and one for postgres.
+
+A collection can be constructed via `NewEtcdCollection` or `NewPostgresCollection`, depending on the backend to be used.  This function must be given:
+ * a template protobuf `Message`, which will be clones for use in serialization and deserialization
+ * a set of fields to create secondary indexes for (used by `GetByIndex` and `WatchByIndex`)
+
+## API
+
+The collection API is split into two means of accessing a collection - `ReadOnly` and `ReadWrite`.  A `ReadOnly` collection does not make any consistency guarantees, while a `ReadWrite` collection is guaranteed to do all reads and writes within a single transaction, although this comes with trade-offs.
+
+### ReadOnly
+
+A `ReadOnlyCollection` is constructed with a context that can be used for cancellation.
+
+Operations:
```

Comment:
This is excellent documentation, but all of it is less discoverable than it would be on the interface methods.  That shows up in autocomplete and godoc.  I don't have strong opinions on how to keep those in sync, but it's worth mentioning.